### ISSUE TITLE
Stalled

### ISSUE THREAD

**Issue by joemck** [miltoncandelero/Boop#20]
I really want this utility to work, it would be way more convenient than servefiles.py.

I select a cia file in Boop 1.3.0, open FBI 2.4.6, select remote install, receive URLs over the network, and click boop files in Boop. FBI displays the number of files I've selected in Boop and I confirm I want it to install them. It then sits at 0/1 files, 0 B/0 B, 0 B/s forever.

Strangely even B will not cancel it and I have to use the home button to escape from FBI. When I close FBI, Boop announces "Booping complete!" Turning off Windows Firewall has no effect on this.

When I use servefiles.py, FBI shows the same zero-bytes-out-of-zero-bytes progress display for under a second, then shows the proper filesize and begins downloading it.

**Comment by miltoncandelero** [comment]
Ok, let's try this:

Select a file with Boop, send it to the 3ds and when it stalls, leave everything hanging, open a web browser on your computer and navigate to

    localhost:8080

You should see a list of files. Try clicking the same file you are trying to send from that list. Does it work?

**Comment by joemck** [comment]
When I do that, it shows me a list of all files in the directory containing the file(s) being booped. When I click one of them, my browser successfully downloads it. I tried putting the file in a folder by itself and giving it a name with no spaces in it. Same result.

Ah-ha! I've identified the issue. After letting it sit for long enough, FBI finally times out and displays this error:
Failed to install from URL.
http://192.168.66.1:8080/nesremix.cia
Result code: 0xD8A0A049
Level: Permanent (27)
Summary: Invalid state (5)
Module: HTTP (40)
Desc: <unknown> (73)

192.168.66.1 is not my IP. It's the IP for VMware's virtual ethernet adapter. I have VMware installed on my PC and it adds some virtual network controllers for networking VMs.

Any possibility of making Boop choose what IP to grab by looking at the default route? Or just an option to see the one it detected and fix it if it's wrong?
